Assets/Scripts/Audio/AudioWaveformDrawer.cs

```diff
@@ -0,0 +1,317 @@
+        public string currentSongTitle => audioSource.clip.name;
+
+        /// <summary>
+        /// Checks if a audio waveform texture has been made.
+        /// </summary>
+        public bool hasActiveWaveform => _waveformTexture != null;
+
+        /// <summary>
+        /// Returns if audio is paused.
+        /// </summary>
+        public bool isPaused { get; private set; }
+
+        /// <summary>
+        /// Time amount for scrubbing through the audio track.
+        /// </summary>
+        public float timeScrubAmount { get; set; } = 2.5f;
+
+        /// <summary>
+        /// Audio source used by the waveform drawer.
+        /// </summary>
+        public AudioSource audioSource { get; private set; }
+
+        /// <summary>
+        /// Event for when the song has been changed. Returns the new song title and the old song title.
+        /// </summary>
+        public Action<string, string> onSongChanged { get; set; }
+
+        /// <summary>
+        /// Event for when the keybinds are shown.
+        /// </summary>
+        public Action onShowKeyBinds { get; set; }  
+
+        private float _playbackSpeed = 10;
+        private float _songWidth = 0;
+
+        private float _audioClipDuration = 0;
+        private float[] _dataSamples = null;
+
+        private Vector2 _waveformPositionOffset = Vector2.zero;
+        private Color[] _textureColors = null;
+
+        private Texture2D _waveformTexture = null;
+
+        private void Awake()
+        {
+            audioSource = GetComponent<AudioSource>();
+        }
+
+        /// <summary>
+        /// Calculates the correct local song time based on a point along the waveform.
+        /// </summary>
+        /// <param name="position">Position to calculate the song time.</param>
+        /// <returns>Song time (float)</returns>
+        public float CalculateSongTimeBasedOnPosition(Vector2 position)
+        {
+            float xPos = position.x - _waveformPositionOffset.x;
+            xPos = Mathf.Clamp(xPos, 0, _songWidth);
+
+            return xPos * _audioClipDuration / _songWidth; ;
+        }
+
+        /// <summary>
+        /// Log all available keybinds to the console.
+        /// </summary>
+        public void LogKeybinds()
+        {
+            Debug.Log("<b>Available Keybinds:</b>");
+            Debug.Log("Pause song key: " + _pauseKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Scrub forward key: " + _forwardKey);
+            Debug.Log("Scrub backward Key: " + _backwardKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Speed up Key: " + _speedUpKey);
+            Debug.Log("Speed down Key: " + _speedDownKey);
+            Debug.Log(" "); //empty line
+
+            Debug.Log("Go to start Key: " + _homeKey);
+            Debug.Log("Go to end Key: " + _endKey);
+
+            Debug.Log(" "); //empty line
+            Debug.Log(" "); //empty line
+            onShowKeyBinds?.Invoke();
+        }
```

Comment:
Maybe make this optional? A debugging boolean for example.

Reply:
This method gets called when the keybinds button is pressed (in custom editor window) so it is already optional